updates to accordion

JulianNymark opened the pull request:
https://github.com/navikt/team-aksel/issues/151

from blue -> gray
no header underline when open

JulianNymark committed:
css updates to accordion

from blue -> gray
no header underline when open

KenAJoh commented on @navikt/core/react/src/accordion/AccordionHeader.tsx:
```diff
@@ -1,4 +1,4 @@
-import { Expand, ExpandFilled } from "@navikt/ds-icons";
+import { Expand } from "@navikt/ds-icons";
```
Kan bytte ut ds-icons med aksel-icons her 
https://aksel.nav.no/ikoner/ChevronDown

JulianNymark replied:
will do!

JulianNymark committed:
modified element tree to mirror that of ExpansionCard

JulianNymark committed:
Accordion wcag tweaks, remove old animation

storybook decorator for styling ("anchor" top of element)

JulianNymark committed:
use button for accordion header, also use :focus-visible

JulianNymark committed:
fallback for css :focus-visible

JulianNymark committed:
bring back onClick (callback)

JulianNymark committed:
changeset commit

JulianNymark committed:
add context to Accordion + style updates

- first element gets border
- variants for background-color "default" and "neutral"
- more heading sizes

JulianNymark committed:
add size prop to accordion + story changes to spacing

JulianNymark committed:
add Default Open to accordion

also rename `--no-fade` to `--no-anim`

JulianNymark committed:
add xsmall heading + stateful --no-anim

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,142 @@
-  background-color: var(--ac-accordion-header-bg-open, var(--ac-accordion-item-bg-open, var(--a-surface-action-subtle)));
-  border-color: var(--ac-accordion-item-border-open, var(--a-surface-action-subtle));
+.navds-accordion__item--open.navds-accordion__item--neutral {
+  background-color: var(--ac-accordion-header-bg-open, var(--ac-accordion-item-bg-open, var(--a-grayalpha-200)));
```
```suggestion
  background-color: var(--ac-accordion-header-bg-open, var(--ac-accordion-item-bg-open, var(--a-surface-neutral-subtle-hover)));
```

KenAJoh commented on @navikt/core/css/tokens.json:
```diff
@@ -14,8 +14,8 @@
-    "--ac-accordion-header-bg": "--ac-surface-transparent",
-    "--ac-accordion-header-bg-hover": "--ac-surface-transparent",
+    "--ac-accordion-header-bg": "--a-surface-transparent",
+    "--ac-accordion-header-bg-hover": "--a-grayalpha-200",
```
```suggestion
    "--ac-accordion-header-bg-hover": "--a-surface-neutral-subtle-hover",
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,142 @@
+}
+
+.navds-accordion__header:hover > .navds-accordion__icon-wrapper {
+  background-color: var(--ac-accordion-header-bg-hover, var(--a-surface-hover));
```
```suggestion
  background-color: var(--ac-accordion-header-bg-hover, var(--a-surface-neutral-subtle-hover));
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,142 @@
   background: var(--ac-accordion-header-bg, var(--a-surface-transparent));
   cursor: pointer;
   border: none;
   border-bottom: 2px solid var(--ac-accordion-header-border, var(--a-border-divider));
```
Foreslår at vi endrer fra å bruke border -> box-shadow for borders her. Størrelser (høyde) vil da matche Figma, og vil kunne se bedre ut med fokus. Unngår at første Accordion-item er 2px høyere enn alle andre også.

```css

.navds-accordion__header {
  ...,
  border: none;
  box-shadow: inset 2px 0 0 0 var(--a-transparent), inset -2px 0 0 0 var(--a-transparent),
    inset 0 -2px 0 0 var(--ac-accordion-header-border, var(--a-border-divider)), inset 0 2px 0 0 var(--a-transparent);
}
```
og
```css
.navds-accordion__item:first-child :where(.navds-accordion__header) {
  box-shadow: inset 2px 0 0 0 var(--a-transparent), inset -2px 0 0 0 var(--a-transparent),
    inset 0 -2px 0 0 var(--ac-accordion-header-border, var(--a-border-divider)),
    inset 0 2px 0 0 var(--ac-accordion-header-border, var(--a-border-divider));
}
```

Kan da legge til border-radius på focus (må dobbelsjekkes med designer)
```css
.navds-accordion__header:focus-visible {
  outline: none;
  box-shadow: var(--a-shadow-focus);
  border-radius: 8px;
}
```

JulianNymark replied:
for å unngå 2px tillegg på border, er det noe vi kunne brukt `box-sizing: border-box;` for?

KenAJoh replied:
Kunne vært en løsning, men får fortsatt denne "visuelle buggen"

border top/bottom
<img width="111" alt="Screenshot 2023-05-02 at 12 07 39" src="https://user-images.githubusercontent.com/26967723/235639291-1f3ce321-4e12-4efb-bccf-247a8f3885d8.png">

border bottom
<img width="151" alt="Screenshot 2023-05-02 at 12 07 42" src="https://user-images.githubusercontent.com/26967723/235639337-94a03c1f-b8b5-47a6-acc1-f9d838929566.png">

JulianNymark replied:
ah right, så hva vi bruker for focus må speiles / brukes likt på "visual element separators". bruker vi border på focus må man bruke border på elementene i en liste, ellers får man dobbelt opp ja, endrer til border:none + box-shadow som du foreslo 👌

JulianNymark replied:
<img width="130" alt="image" src="https://user-images.githubusercontent.com/819074/235643412-ded4bc77-e240-4f6b-870c-9acd8236e51e.png">


litt special casing må legges til for `border-radius` på focus, men skal sjekke med Sjur :) (det er kanskje noe vi har på andre `focus`ede elementer?

KenAJoh replied:
Accordion er en litt special case siden den bare har 90deg hjørner, så har det ikke på andre elementer som jeg husker. Men har vært en kritikk av Accordion vår at den virker veldig boksette og litt overveldende med den veldig store fokusmarkeringen + ingen border-radius

KenAJoh commented on @navikt/core/react/src/accordion/Accordion.tsx:
```diff
@@ -14,16 +15,55 @@ interface AccordionComponent
 
 export interface AccordionProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * @default 'primary'
```
```suggestion
   * @default 'default'
```

KenAJoh commented on @navikt/core/react/src/accordion/Accordion.tsx:
```diff
@@ -14,16 +15,55 @@ interface AccordionComponent
+  /**
+   * @default 'large'
+   */
+  headingsize?: "large" | "medium" | "small" | "xsmall";
```
Vi bruker for det meste camelCase for propnavn 🐫 
```suggestion
  headingSize?: "large" | "medium" | "small" | "xsmall";
```

JulianNymark replied:
Ja, jeg hadde først lyst til å bruke camelCase, men endret det pga det ga utslag fra React i console

<img width="643" alt="image" src="https://user-images.githubusercontent.com/819074/235634847-2ac4a807-0254-40fd-aeb2-a2b41f4c2fd8.png">

🤔 , camelcase gir mer mening, skal undersøke feilen litt mer

JulianNymark replied:
kanskje det bare var storybook props faktisk

KenAJoh replied:
ja tror det er denne i storybook `<Accordion {...props}>` + nItems i ``` args: {
    controlled: false,
    nItems: 2,
    variant: "default",
    headingsize: "medium",
    size: "medium",
  },```

KenAJoh commented on @navikt/core/react/src/accordion/Accordion.tsx:
```diff
@@ -14,16 +15,55 @@ interface AccordionComponent
+          size,
+        }}
+      >
+        <section
```
Ser ut som det er non-standard å wrappe Accordion i `section` som default. Kanskje noe vi kan tilby med OverridableComponent, men med standard satt til `div`?

KenAJoh commented on @navikt/core/react/src/accordion/AccordionContext.tsx:
```diff
@@ -0,0 +1,15 @@
+  variant?: "default" | "neutral";
+  headingsize?: "large" | "medium" | "small" | "xsmall";
+  size?: "large" | "medium" | "small";
+  openitems?: number[];
```
```suggestion
```

KenAJoh commented on @navikt/core/react/src/accordion/AccordionContext.tsx:
```diff
@@ -0,0 +1,15 @@
+  variant: "default",
+  headingsize: "small",
+  size: "medium",
+  openitems: [],
```
```suggestion
```

KenAJoh commented on @navikt/core/react/src/accordion/AccordionHeader.tsx:
```diff
@@ -1,11 +1,12 @@
 
 export interface AccordionHeaderProps
-  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  extends React.HTMLAttributes<HTMLButtonElement> {
```
```suggestion
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
```

KenAJoh commented on @navikt/core/react/src/accordion/AccordionHeader.tsx:
```diff
@@ -18,43 +19,40 @@ export type AccordionHeaderType = React.ForwardRefExoticComponent<
+          itemContext.toggleOpen();
+          onClick && onClick(e);
+        }}
+        aria-expanded={itemContext.open}
```
Tror type button må være med for å unngå feil ved bruk i <form>
```suggestion
        aria-expanded={itemContext.open}
        type="button"
```

JulianNymark replied:
ved bruk i forms?

KenAJoh replied:
Ikke noe vi oppmuntrer til, men er gjengangere at komponentene blir brukt inne i `<form>` (eks https://nav-it.slack.com/archives/C7NE7A8UF/p1683014658766199).

KenAJoh commented on @navikt/core/react/src/accordion/AccordionHeader.tsx:
```diff
@@ -18,43 +19,40 @@ export type AccordionHeaderType = React.ForwardRefExoticComponent<
+        <div className="navds-accordion__icon-wrapper">
+          <ChevronDownIcon
+            className="navds-accordion__header-chevron"
+            title="Vis mer"
```
For accordion vil det være mye støy for skjermleser hvis hjelpetekst blir lest opp for hvert item.
<img width="659" alt="Screenshot 2023-05-02 at 11 50 18" src="https://user-images.githubusercontent.com/26967723/235635172-96a7f17e-558b-4327-9779-5d5883b7b541.png">

```suggestion
      aria-hidden
```

KenAJoh commented on @navikt/core/react/src/expansion-card/ExpansionCard.tsx:
```diff
@@ -113,7 +113,7 @@ export const ExpansionCard = forwardRef<HTMLDivElement, ExpansionCardProps>(
             {
               "navds-expansioncard--open": open ?? _open,
-              "navds-expansioncard--no-fade": !shouldFade.current,
+              "navds-expansioncard--no-anim": !shouldFade.current,
```
Har ikke noen sterke meninger om dette, men liker selv å skrive navnene ut der man kan `navds-expansioncard--no-animation`

JulianNymark replied:
ja, jeg tok en kjap en der (ville dukke opp i PR anyways, så det _bare_ var i ExpansionCard du hadde lagt det til, så her var det mulighet for en mer generisk konvensjon om jeg renamet ExpansionCard). Men skal endre til animation 👍

KenAJoh commented on @navikt/core/react/src/accordion/accordion.stories.tsx:
Ønsker også en story som lister ut alle `size`-valgene, og en med alle `headingSize`-valgene (kanskje en med de også kombinert?). Da får vi de som snapshot i Chromatic for fremtidig testing 🙏 🧪

KenAJoh commented on @navikt/core/react/src/accordion/AccordionItem.tsx:
```diff
@@ -38,26 +45,38 @@ const AccordionItem: AccordionItemType = forwardRef(
           }}
         >
-          {children}
+          <div>{children}</div>
```
Trengs denne div-wrapperen? Kan hende det er noe CSS som jeg misset tilhørende. Tror kanskje den brekker denne CSS-en nå om så 
```
.navds-accordion__item--open > .navds-accordion__header:hover + * .navds-accordion__content {
  border-color: var(--ac-accordion-content-border-open, var(--a-border-default));
}
```

KenAJoh reviewed (changes requested):
Bra jobba! 🙌 Har gått gjennom å funnet noen småting som kan ses på. Kan også legge til kode-eksempler for Aksel.nav.no i samme PR:
https://github.com/navikt/aksel/tree/main/aksel.nav.no/website/pages/eksempler/accordion
[Howto](https://github.com/navikt/aksel/blob/main/aksel.nav.no/website/pages/eksempler/README.md)

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,142 @@
-.navds-accordion__expand-icon--filled {
-  display: none;
+.navds-accordion__item--open :where(.navds-accordion__header-chevron) {
+  transform: translateY(0) rotate(180deg);
```
Rotere pil på "insiden"
```suggestion
  transform: translateY(0) rotate(-180deg);
```

JulianNymark replied:
Confused here, jeg får rar "360" spin on hover om jeg lager denne endringen

JulianNymark replied:
aaah nvm, henger sammen med under... got it... ja, retningen den spinner

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,142 @@
-.navds-accordion__header:hover > .navds-accordion__expand-icon.navds-accordion__expand-icon--filled {
-  display: inherit;
+.navds-accordion__item--open :where(.navds-accordion__header):hover :where(.navds-accordion__header-chevron) {
+  transform: translateY(-1px) rotate(180deg);
```
```suggestion
  transform: translateY(-1px) rotate(-180deg);
```

JulianNymark committed:
PR comment fixes

JulianNymark committed:
css selector fixes for border / box-shadow

JulianNymark committed:
use div over section for accordion

JulianNymark committed:
css from first -> last, + more tokens

JulianNymark committed:
remove border in favor of box-shadow

JulianNymark committed:
fix last-child bug with box-shadow

JulianNymark committed:
update demos

JulianNymark committed:
change demo names to not have whitespace

KenAJoh committed:
Merge branch 'main' into update-accordion

KenAJoh committed:
:art: More stories

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
+.navds-accordion__header:hover > .navds-accordion__header-chevron {
+  background-color: var(--ac-accordion-header-bg-hover, var(--a-surface-hover));
+  border-radius: var(--a-border-radius-medium);
+  height: 24px;
```
```suggestion
  height: 1.5rem;
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
   align-items: flex-start;
   gap: var(--a-spacing-2);
-  padding: 14px var(--a-spacing-3) var(--a-spacing-3);
+  padding: var(--a-spacing-3) var(--a-spacing-3);
```
```suggestion
  padding: var(--a-spacing-3);
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
+}
+
+.navds-accordion--medium .navds-accordion__header {
+  padding: var(--a-spacing-3) var(--a-spacing-3);
```
```suggestion
  padding: var(--a-spacing-3);
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
-  color: var(--ac-accordion-header-text-hover, var(--a-text-action-on-action-subtle));
-  border-color: var(--ac-accordion-header-border-hover, var(--a-border-default));
-  background: var(--ac-accordion-header-bg-hover, var(--a-surface-transparent));
+  background: var(--ac-accordion-header-bg-hover, var(--a-grayalpha-200));
```
```suggestion
  background: var(--ac-accordion-header-bg-hover, var(--a-surface-neutral-subtle-hover));
```

KenAJoh replied:
For bedre backwards compat
```suggestion
  color: var(--ac-accordion-header-text-hover, inherit);
  background: var(--ac-accordion-header-bg-hover, var(--a-grayalpha-200));
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
+.navds-accordion__header > .navds-accordion__icon-wrapper {
+  display: grid;
+  place-content: center;
+  cursor: pointer;
```
```suggestion
```

KenAJoh commented on @navikt/core/css/accordion.css:
```diff
@@ -5,70 +5,157 @@
+  display: grid;
+  place-content: center;
+  cursor: pointer;
+  margin: 0;
```
Tror en del kan fjernes her, henger kanskje igjen fra når det var en <button>

KenAJoh commented on @navikt/core/css/tokens.json:
```diff
@@ -24,7 +24,8 @@
     "--ac-accordion-content-border": "--a-border-divider",
-    "--ac-accordion-content-border-open": "--a-border-default"
+    "--ac-accordion-content-border-open": "--a-border-default",
+    "--ac-accordion-border-radius": "--a-border-radius-medium"
```
Ser ikke ut som denne blir brukt noen plass

KenAJoh committed:
Update @navikt/core/css/accordion.css

KenAJoh committed:
Update @navikt/core/css/accordion.css

KenAJoh committed:
:lipstick: Statusikoner er nå tydeligere og mer konsistente. (#1959)

github-actions[bot] committed:
Version Packages (#1960)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

JulianNymark committed:
fix review comments

- rm unused tokens
- rm unused css

AndreasDybdahl committed:
Update ExpansionCardDescription.tsx (#1962)

KenAJoh committed:
Merge branch 'main' into update-accordion

KenAJoh committed:
:lipstick: Subtle animation fade

KenAJoh committed:
:recycle: Refactred parts of the CSS for tokens

KenAJoh committed:
Merge branch 'v3-major' into update-accordion
